docs/src/main/tut/typeclasses/lawtesting.md

```diff
@@ -0,0 +1,146 @@
+# Law testing
+
+[Laws](https://typelevel.org/cats/typeclasses.html#laws) are an important part of cats.
+They help us enforce constraints that we usually assume to be true and give us guarantees that our code does what we think it does.
+Manually testing each data type to adhere to each of its type class instances can be quite cumbersome.
+Thankfully cats comes with the `cats-testkit`, which makes it a lot easier to write tests for type class instances using laws.
+
+
+## Getting started
+
+First up, you will need to specify dependencies on `cats-laws` and `cats-testkit` in your `build.sbt` file.
+To make things easier, we'll also include the `scalacheck-shapeless` library in this tutorial, so we don't have to manually write instances for ScalaCheck's `Arbitrary`.
+
+```scala
+libraryDependencies ++= Seq(
+  "org.typelevel" %% "cats-laws" % "1.0.0-MF" % Test,
+  "org.typelevel" %% "cats-testkit" % "1.0.0-MF"% Test,
+  "com.github.alexarchambault" %% "scalacheck-shapeless_1.13" % "1.1.5" % Test
+)
+```
+
+## Example: Testing a Functor instance
+
+We'll begin by creating a data type and its Functor instance.
+```tut:book
+import cats._
+
+sealed trait Tree[+A]
+case object Leaf extends Tree[Nothing]
+case class Node[A](p: A, left: Tree[A], right: Tree[A]) extends Tree[A]
+
+object Tree {
+  implicit val functorTree: Functor[Tree] = new Functor[Tree] {
+    def map[A, B](tree: Tree[A])(f: A => B) = tree match {
+      case Leaf => Leaf
+      case Node(p, left, right) => Node(f(p), map(left)(f), map(right)(f))
+    }
+  }
+}
+```
+We will also need to create an `Eq` instance, as most laws will need to compare values of a type to properly test for correctness.
+For simplicity we'll just use `Eq.fromUniversalEquals`:
+
+```tut:book
+implicit def eqTree[A: Eq]: Eq[Tree[A]] = Eq.fromUniversalEquals
+```
+
+Then we can begin to write our law tests. Start by creating a new class in your `test` folder and inheriting from `cats.tests.CatsSuite`.
+`CatsSuite` extends the standard ScalaTest `FunSuite` as well as `Matchers`.
+Furthermore it also pulls in all of cats instances and syntax, so there's no need to import from `cats.implicits._`.
+
+```tut:book
+import cats.tests.CatsSuite
+
+class TreeLawTests extends CatsSuite {
+
+}
+```
+
+The key to testing laws is the `checkAll` function, which takes a name for your test and a Discipline ruleset.
+Cats has defined rulesets for all type class laws in `cats.laws.discipline.*`.
+
+So for our example we will want to import `cats.laws.discipline.FunctorTests` and call `checkAll` with it.
+Before we do so, however,
+we will have to bring our instances into scope as well as the derived `Arbitrary` instances from `scalacheck-shapeless`:
+
+
+```tut:silent
+import Tree._
+```
+```scala
+import org.scalacheck.Shapeless._
+```
+
+```tut:invisible
+import org.scalacheck.{Arbitrary, Gen}
+implicit def arbFoo[A: Arbitrary]: Arbitrary[Tree[A]] =
```

Comment:
so this is the one that is actually in use, not the one generated by scalacheck-shapeless?

Reply:
Yeah, I realize it's not an optimal solution, but it's the best I came up with. If you have a better idea, please do tell :)

Reply:
i think the spirit of our documentation is that people can just copy code and it runs. This breaks that assumption. why scalacheck-shapeless didn't work?

Reply:
I wouldn't know how to include a foreign module for a tut doc 🤔. I guess it would be possible to add it as a dependency, but not sure if that's a good idea for just one part of the documentation?

Reply:
I am inclined to not pretend to use it in the documentation. Make the arbitrary instance visible and add a note that you can use scalacheck-shapeless to auto derive this instance. 
BTW, could we try avoid the `asInstance` in the arbitrary instance?

Reply:
I got it to work by adding this single line to `docs` in the build.sbt:

```
.settings(libraryDependencies += "com.github.alexarchambault" %% "scalacheck-shapeless_1.13" % "1.1.5" % Compile)
```

tut then compiles the whole thing without a hitch.

Reply:
I still prefer making this visible and mentioned the alternative solution from `scalacheck-shapeless` in a note.